cassiopeia/simulator/Cas9LineageTracingDataSimulator.py

```diff
@@ -132,42 +144,70 @@ def __init__(
         self.size_of_cassette = size_of_cassette
         self.number_of_cassettes = number_of_cassettes
         self.collapse_sites_on_cassette = collapse_sites_on_cassette
+        number_of_characters = size_of_cassette * number_of_cassettes
 
         if isinstance(mutation_rate, float):
             if mutation_rate < 0:
                 raise DataSimulatorError(
                     "Mutation rate needs to be" " non-negative."
                 )
-            number_of_characters = size_of_cassette * number_of_cassettes
             self.mutation_rate_per_character = [
                 mutation_rate
             ] * number_of_characters
-        else:
-            if len(mutation_rate) != (
-                self.number_of_cassettes * self.size_of_cassette
-            ):
+        elif isinstance(mutation_rate, list):
+            if len(mutation_rate) == number_of_characters:
+                self.mutation_rate_per_character = mutation_rate
+            elif len(mutation_rate) == self.size_of_cassette:
+                self.mutation_rate_per_character = (
+                    mutation_rate * self.number_of_cassettes
+                )
+            else:
                 raise DataSimulatorError(
                     "Length of mutation rate array is not"
-                    " the same as the number of characters."
+                    " the same as the number of characters"
+                    " or the number of cassettes."
                 )
 
             if np.any(np.array(mutation_rate) < 0):
                 raise DataSimulatorError(
                     "Mutation rate needs to be" " non-negative."
                 )
-            self.mutation_rate_per_character = mutation_rate[:]
+        else:
+            raise DataSimulatorError(
+                "Mutation rate needs to be a float or a list."
+            )
+            
 
         if state_priors is None:
             self.number_of_states = number_of_states
             self.state_generating_distribution = state_generating_distribution
+            self.mutation_priors_per_character = None
         else:
-            Z = np.sum([v for v in state_priors.values()])
-            if not math.isclose(Z, 1.0):
-                raise DataSimulatorError("Mutation priors do not sum to 1.")
-            self.number_of_states = len(state_priors)
+            if isinstance(state_priors, dict):
+                self.mutation_priors_per_character = [
+                    state_priors
+                ] * number_of_characters
+            else:
+                if len(state_priors) == number_of_characters:
+                    self.mutation_priors_per_character = state_priors
+                elif len(state_priors) ==  self.size_of_cassette:
```

Comment:
can we have the same problem above if `state_prior` is a numpy array?

Reply:
Yes. Good catch!